trilium_alchemy/core/note/note.py

```diff
@@ -962,14 +952,46 @@ def __getitem__(self, key: str | int) -> str | Note:
             else:
                 return attr.value
 
-    def __setitem__(self, key: str | int, value_spec: ValueSpec):
+    def __setitem__(self, key: str, value_spec: ValueSpec):
         """
-        Create or update attribute with provided name.
+        Create or update attribute with provided name or index.
+
+        :param key: Attribute name
+        :param value_spec: Attribute value
         """
         self.attributes[key] = value_spec
 
-    def __contains__(self, key: str) -> bool:
-        return key in self.attributes._name_map
+    def __hash__(self):
+        return hash(self.note_id)
+
+    def __eq__(self, other):

```

Comment:
Was this needed? There can't be 2 instances of `Note` with the same `note_id` due to the cache. `Note.__new__` will return the existing object with that `note_id`.

Reply:
__eq__ was needed to meet the Mapping contract.

Reply:
I see. Then we should be able to just use:

```python
    return self is other
```

This should be what Python was automatically doing before we subclassed `MutableMapping`.